### PR TITLE
fix: remove docker installation logic and checks in VHD build, remove unused 16.04 post VHD build test

### PR DESCRIPTION
**Pull request by ganeshkumar5699**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
- Removes unused docker specific installation checks/steps during VHD build
- Prevents the need to update components that are specific to Docker, and makes installation steps more readable
- Clean up unused 16.04 post VHD build check

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/packer/install-dependencies.sh [hunk 1]
```diff
@@ -189,23 +185,14 @@ if [[ $OS == $UBUNTU_OS_NAME && $(isARM64) != 1 ]]; then  # no ARM64 SKU with GP
...
+    exit 1
   fi
 fi
 
 if [ "${CONTAINER_RUNTIME:=}" == "containerd" ]; then
     systemctlEnableAndStart containerd-monitor.timer || exit $ERR_SYSTEMCTL_START_FAIL
-else
-    systemctlEnableAndStart docker-monitor.timer || exit $ERR_SYSTEMCTL_START_FAIL
 fi
-
```
Comment (cameronmeissner) [inline vhdbuilder/packer/install-dependencies.sh:208]: maybe also fail hard here if container runtime isn't containerd?

File: vhdbuilder/packer/install-dependencies.sh [hunk 2]
```diff
@@ -269,10 +256,7 @@ watcherStaticImg=${watcherBaseImg//\*/static}
 # can't use cliTool because crictl doesn't support retagging.
 if [[ "${CONTAINER_RUNTIME}" == "containerd" ]]; then
     retagContainerImage "ctr" ${watcherFullImg} ${watcherStaticImg}
-else
-    retagContainerImage "docker" ${watcherFullImg} ${watcherStaticImg}
 fi
-
```
Comment (cameronmeissner) [inline vhdbuilder/packer/install-dependencies.sh:275]: diddo

**Review by cameronmeissner — APPROVED** [review]

File: vhdbuilder/packer/install-dependencies.sh (hunk 1 above)
Comment (anujmaheshwari1) [reply vhdbuilder/packer/install-dependencies.sh:208]: following from what Cam said, we seem to be using
`echo "Unsupported container runtime"
exit 1`

for cases where cri isnt containerd. maybe make a helper function that has those 2 lines and call it in all the else statements?

Comment (ganeshkumar5699) [reply vhdbuilder/packer/install-dependencies.sh:208]: L145 already will fail, if runtime is not containerd. Having that check earlier prevents the need from repeating it right?

Comment (alexeldeib) [reply vhdbuilder/packer/install-dependencies.sh:208]: should we just do this once at the very beginning of the build and stop bothering to check elsewhere?

Comment (cameronmeissner) [reply vhdbuilder/packer/install-dependencies.sh:188]: > L145 already will fail, if runtime is not containerd. Having that check earlier prevents the need from repeating it right?

I probably just missed that :D

**Review by cameronmeissner — APPROVED** [review]

**Review by alexeldeib — APPROVED** [review]
nice!

**Pull request merged**